usecases/tasks_handler.go

```
@@ -0,0 +1,251 @@
+	rawTxs, err := t.dbRepo.QueryRawTxsByTaskTransformID(taskId)
+	if err != nil {
+		return common.NewError(common.ERROR_INTERNAL, err)
+	}
+
+	var onchainTxList []*entities.OnchainTransaction
+
+	var accountList []*entities.Account
+
+	var accountTxs []*entities.AccountTx
+
+	for _, rawTransaction := range rawTxs {
+
+		if strings.HasPrefix(rawTransaction.ChaincodeProposalInput, "Transaction.") {
+			newTransactions, newAccountTxs, err := t.rawTxToTransactions(rawTransaction, taskId)
+			if err != nil {
+				return err
+			}
+			onchainTxList = append(onchainTxList, newTransactions...)
+			accountTxs = append(accountTxs, newAccountTxs...)
+
+		}
+
+		if strings.HasPrefix(rawTransaction.ChaincodeProposalInput, "Account.") {
+			newAccounts, err := t.rawTxToAccounts(rawTransaction, taskId)
+			if err != nil {
+				return err
+			}
+			accountList = append(accountList, newAccounts...)
+
+		}
+
+	}
+	err = t.dbRepo.SaveTransformData(taskId, onchainTxList, accountList, accountTxs)
+	if err != nil {
+		return common.NewError(common.ERROR_INTERNAL, err)
+	}
+	return nil
+}
+
+func (t *taskHandler) calculateAccountDelta(taskId int64) common.BaseError {
+	return nil
+}
+
+func (t *taskHandler) rawTxToTransactions(rawTransaction *entities.RawTransaction, taskId int64) ([]*entities.OnchainTransaction, []*entities.AccountTx, common.BaseError) {
+	var onchainTxList []*entities.OnchainTransaction
+	var accountTxsList []*entities.AccountTx
+
+	response, err := contractpb.NewTransactionResponse(rawTransaction.Payload)
+	if err != nil {
+		t.logger.Errorf("Create new transaction response error %s", err.Error())
+		return nil, nil, common.NewUnknownError(err)
+	}
+
+	for _, attribute := range response.Transaction.Attributes {
+
+		preDay, err := time.Parse(common.YYYYMMDD, attribute.OffchainTime.PreDay)
+		if err != nil {
+			t.logger.Errorf("Error when parse preday %v error %s", attribute.OffchainTime.PreDay, err.Error())
+			// return nil, nil, common.NewUnknownError(err)
```

Update the `rawTxToTransactions` function to return errors immediately after logging them to prevent silent failures [important]